console/src/view/mod.rs

```diff
@@ -0,0 +1,61 @@
+use crate::input;
+use tui::layout;
+
+mod task;
+
+pub(crate) enum View {
+    /// The table list of all tasks.
+    TasksList,
+    /// Inspecting a single task instance.
+    TaskInstance(self::task::TaskView),
+}
+
+impl View {
+    pub(crate) fn update_input(&mut self, event: input::Event, tasks: &mut crate::tasks::State) {
+        match self {
+            View::TasksList => {
+                // The enter key changes views, so handle here since we can
+                // mutate the currently selected view.
+                match event {
+                    input::Event::Key(input::KeyEvent {
+                        code: input::KeyCode::Enter,
+                        ..
+                    }) => {
+                        if let Some(task) = tasks.selected_task().upgrade() {
+                            *self = View::TaskInstance(self::task::TaskView::new(task));
+                        }
+                    }
+                    _ => {
+                        // otherwise pass on to view
+                        tasks.update_input(event);
+                    }
+                }
+            }
+            View::TaskInstance(view) => view.update_input(event),
```

Comment:
i feel like it could be good to have a way to get back out of the single task view and return to the task list...the escape key seems like a natural choice, or pressing enter again?